.azure_pipelines/docker_images/ubuntu_1804_py_38_cuda_102_cudnn_8_dev/Dockerfile

```diff
@@ -0,0 +1,52 @@
+# This is the Dockerfile for an image that is ready to build PyTorch from source.
+# PyTorch is not yet downloaded nor installed.
+# Specification: Ubuntu 18.04, CUDA 10.2, cuDNN 8.1.0
+# The ARGs below are currently the only differences between other Ubuntu CUDA builds.
+#   PYTHON_VERSION
+#   TORCH_CUDA_ARCH_LIST_VAR
+
+ARG BASE_IMAGE=nvidia/cuda:10.2-cudnn8-devel-ubuntu18.04
```

Comment:
Most of these docker images look almost exactly the same, do we think we can consolidate these as a single Dockerfile that just has separate stages / args that make it so there’s not a maintenance cost of multiple Dockerfiles?

Reply:
That is true, as I said above I will be consolidating the Dockerfiles and the Windows env yamls.

Reply:
I've consolidated the Dockerfiles into Ubuntu CPU and GPU Dockerfiles, where with build args the needed Dockerfile can be generated (given base image, CUDN/cuDNN config/ Py Version, etc...)